test/cpp/interop/interop_client.h

```diff
@@ -35,17 +38,25 @@ typedef std::function<void(const InteropClientContextInspector&,
                            const SimpleRequest*, const SimpleResponse*)>
     CheckerFn;
 
-typedef std::function<std::shared_ptr<Channel>(void)> ChannelCreationFunc;
+typedef std::function<std::shared_ptr<Channel>()> ChannelCreationFunc;
+
+typedef std::function<std::shared_ptr<Channel>(
+    std::function<void(ChannelArguments*)>)>
+    ChannelCreationFuncWithCustomArgs;
 
 class InteropClient {
  public:
   /// If new_stub_every_test_case is true, a new TestService::Stub object is
   /// created for every test case
-  /// If do_not_abort_on_transient_failures is true, abort() is not called in
-  /// case of transient failures (like connection failures)
+  /// If do_not_abort_on_transient_failures is true, abort() is not called
+  /// in case of transient failures (like connection failures)
   explicit InteropClient(ChannelCreationFunc channel_creation_func,
                          bool new_stub_every_test_case,
                          bool do_not_abort_on_transient_failures);
+  /// Same as ctor above, allows customizing channel arguments
+  explicit InteropClient(
```

Comment:
Instead of adding a second ctor, why not just change the existing callers to use the new parameter?

Reply:
I only see 3 other callers, and they look trivial to change.  Avoiding this doesn't seem to save much time.  And unless you plan to write a subsequent PR to clean this up, this is just leaving unnecessary technical debt.

The down-side of this approach is that any caller that uses the old interface will not get the `LoadReportTracker` channel arg.

Reply:
Every caller also needs to be retrofitted with the JSON command line arg and such. I understand this is not necessary for current pull request and figuring out all the build dependencies and such delays this pull request.

I removed the second ctor but would add JSON config flag in a separate PR if needed. I feel like some refactoring effort is needed to reduce code duplication.

Reply:
I don't think the JSON command line flag is needed in the other binaries, so I don't think there's any need for a follow-on PR here.  This looks great!